File: Challenge 1/Challenge 1.c
#include <stdio.h>
#include <stdlib.h>

#define GRAND_CLOCK_HOUR 15
#define GRAND_CLOCK_MINUTE 0

// Function to calculate the time difference in minutes
int calculate_time_difference(int clock_hour, int clock_minute) {
    int grand_clock_total_minutes = GRAND_CLOCK_HOUR * 60 + GRAND_CLOCK_MINUTE;
    int clock_total_minutes = clock_hour * 60 + clock_minute;
    return clock_total_minutes - grand_clock_total_minutes;
}

int main() {
    // Clock times around town
    int clock_hours[] = {14, 15, 15, 14, 15};
    int clock_minutes[] = {45, 5, 0, 40, 2};
    int num_clocks = sizeof(clock_hours) / sizeof(clock_hours[0]);

    // Array to store the time differences
    int time_differences[num_clocks];

    // Calculate the time differences for each clock
    for (int i = 0; i < num_clocks; i++) {
        time_differences[i] = calculate_time_difference(clock_hours[i], clock_minutes[i]);
    }

    // Output the results
    printf("Time differences (in minutes):\n");
    for (int i = 0; i < num_clocks; i++) {
        printf("Clock %d: %d minutes\n", i + 1, time_differences[i]);
    }

    return 0;
}

File: Challenge 1/Challenge 1.py
# Define the Grand Clock Tower time
grand_clock_time = "15:00"

# Define the clock times around town
clock_times = ["Shivanshu", "15:05", "15:00", "14:40"]

def calculate_time_difference(grand_time, clock_times):
    # Convert the Grand Clock Tower time to minutes
    grand_hours, grand_minutes = map(int, grand_time.split(":"))
    grand_total_minutes = grand_hours * 60 + grand_minutes

    # Initialize an array to store the time differences
    time_differences = []

    # Calculate the difference for each clock
    for clock_time in clock_times:
        try:
            clock_hours, clock_minutes = map(int, clock_time.split(":"))
            clock_total_minutes = clock_hours * 60 + clock_minutes
            time_difference = clock_total_minutes - grand_total_minutes
            time_differences.append(time_difference)
        except ValueError:
            # Skip invalid time formats
            continue

    return time_differences

# Calculate the time differences
differences = calculate_time_difference(grand_clock_time, clock_times)

# Output the result
print("Time differences (in minutes):", differences)